R/diagnostic_test.R

```diff
@@ -0,0 +1,198 @@
+  if (any(!is.na(names(proxy_config)))) {
+    dfeR::toggle_message("Found proxy settings:", verbose = verbose)
+    paste(names(proxy_config), "=", proxy_config, collapse = "\n") |>
+      toggle_message(verbose = verbose)
+    if (clean) {
+      proxy_args <- proxy_config |>
+        lapply(function(list) {
+          NULL
+        })
+      rlang::inject(git2r::config(!!!proxy_args, global = TRUE))
+      message("FIXED: Git proxy settings have been cleared.")
+    } else {
+      message("FAIL: Git proxy setting have been left in place.")
+    }
+  } else {
+    message("PASS: No proxy settings found in your Git configuration.")
+    proxy_config <- as.list(rep("", length(proxy_setting_names))) |>
+      stats::setNames(proxy_setting_names)
+  }
+  return(proxy_config)
+}
+
+#' Check GITHUB_PAT setting
+#'
+#' @description
+#' If the GITHUB_PAT keyword is set, then it can cause issues with R installing
+#' packages from GitHub (usually with an error of "ERROR \[curl: (22) The
+#' requested URL returned error: 401\]"). This script checks whether the keyword
+#' is set and can then clear it (if clear=TRUE).
+#' The user will then need to identify where the "GITHUB_PAT" variable is being
+#' set from and remove it to permanently fix the issue.
+#'
+#' @inheritParams check_proxy_settings
+#'
+#' @return List object containing the github_pat keyword content.
+#' @export
+#'
+#' @examples
+#' check_github_pat()
+check_github_pat <- function(
+    clean = FALSE,
+    verbose = FALSE) {
+  github_pat <- Sys.getenv("GITHUB_PAT")
+  # Replace above to remove non alphanumeric characters when run on GitHub
+  # Actions
+  cat("==================================")
+  if (!is.na(github_pat)) {
+    message(
+      "FAIL: GITHUB_PAT is set to ",
+      github_pat,
+      ". This may cause issues with installing packages from GitHub",
+      " such as dfeR and dfeshiny."
+    )
+    if (clean) {
+      message("Clearing GITHUB_PAT keyword from system settings.")
+      Sys.unsetenv("GITHUB_PAT")
+      message(
+        "This issue may recur if you have some software that is",
+        "initialising the GITHUB_PAT keyword automatically."
+      )
+    }
+  } else {
+    message("PASS: The GITHUB_PAT system variable is clear.")
+  }
+  return(list(GITHUB_PAT = github_pat))
+}
+
+#' Check renv download method
+#'
+#' @description
+#' The renv package can retrieve packages either using curl or wininet, but
+#' wininet doesn't work from within the DfE network. This script checks for
+#' the parameter controlling which of these is used (RENV_DOWNLOAD_METHOD) and
+#' sets it to use curl.
+#'
+#' @param renviron_file Location of .Renviron file. Default: ~/.Renviron
+#' @inheritParams check_proxy_settings
+#'
+#' @return List object containing RENV_DOWNLOAD_METHOD
+#' @export
+#'
+#' @examples
+#' check_renv_download_method()
+check_renv_download_method <- function(
+    renviron_file = "~/.Renviron",
+    clean = FALSE,
+    verbose = FALSE) {
```

Comment:
Feel like the `verbose = FALSE` option should only print the success message, feels a bit too verbose printing the variable as well as the message?

![image](https://github.com/user-attachments/assets/7eb022b1-0564-474f-bc8e-d54d12208f33)